patches/server/0032-Add-configurable-entity-despawn-distances.patch

```diff
@@ -5,23 +5,44 @@ Subject: [PATCH] Add configurable entity despawn distances
 
 
 diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
-index bf905bc1c4918412e3c324a62d2915b57c40d2cc..51b33ea29af0f4010dfb8a1a7503cb7ca463209e 100644
+index bf905bc1c4918412e3c324a62d2915b57c40d2cc..df11e6548a43f0ce81f8f434ac6c93f9fbd4d398 100644
 --- a/src/main/java/net/minecraft/world/entity/Mob.java
 +++ b/src/main/java/net/minecraft/world/entity/Mob.java
-@@ -866,14 +866,14 @@ public abstract class Mob extends LivingEntity implements EquipmentUser, Leashab
+@@ -865,20 +865,27 @@ public abstract class Mob extends LivingEntity implements EquipmentUser, Leashab
+             Player entityhuman = this.level().getNearestPlayer(this, -1.0D);
  
              if (entityhuman != null) {
-                 double d0 = entityhuman.distanceToSqr((Entity) this);
+-                double d0 = entityhuman.distanceToSqr((Entity) this);
 -                int i = this.getType().getCategory().getDespawnDistance();
-+                int i = this.level().paperConfig().entities.spawning.despawnRanges.get(this.getType().getCategory()).hard(); // Paper - Configurable despawn distances
-                 int j = i * i;
+-                int j = i * i;
++                // Paper start - Configurable despawn distances
++                io.papermc.paper.configuration.WorldConfiguration.Entities.Spawning.DespawnRange despawnRanges = this.level().paperConfig().entities.spawning.despawnRanges.get(this.getType().getCategory());
  
-                 if (d0 > (double) j && this.removeWhenFarAway(d0)) {
-                     this.discard(EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
-                 }
+-                if (d0 > (double) j && this.removeWhenFarAway(d0)) {
+-                    this.discard(EntityRemoveEvent.Cause.DESPAWN); // CraftBukkit - add Bukkit remove cause
+-                }
++                double xzDistance = entityhuman.distanceToSqr(this.getX(), entityhuman.getY(), this.getZ());
```

Comment:
Add a squared suffix where applicable here and below, else the variables can be confused easily